packages/portalnetwork/src/subprotocols/history/history.ts

```diff
@@ -277,41 +279,51 @@ export class HistoryProtocol extends BaseProtocol {
       this.logger(`Block number ${blockNumber} is higher than current known chain height`)
       return
     }
-    const blockIndex = blockNumber % EPOCH_SIZE
-    const historicalEpochIndex = Math.floor(blockNumber / EPOCH_SIZE)
-    const epochRootHash = this.accumulator.historicalEpochs[historicalEpochIndex]
-
-    let blockHash
-    if (!epochRootHash) {
-      blockHash = toHexString(this.accumulator.currentEpoch[blockIndex].blockHash)
-    } else {
-      let epoch
+    if (blockNumber > 8192 * this.accumulator.historicalEpochs.length) {
+      const blockIndex = blockNumber % EPOCH_SIZE
+      const blockHash = toHexString(this.accumulator.currentEpoch[blockIndex].blockHash)
+      this.logger(`Blockhash found for BlockNumber ${blockNumber}: ${blockHash}`)
       try {
-        const encodedEpoch = await this.client.db.get(
-          getHistoryNetworkContentId(
-            1,
-            HistoryNetworkContentTypes.EpochAccumulator,
-            toHexString(epochRootHash)
-          )
-        )
-        epoch = EpochAccumulator.deserialize(fromHexString(encodedEpoch))
-      } catch {
-        const lookup = new ContentLookup(
-          this,
-          HistoryNetworkContentKeyUnionType.serialize({
-            selector: HistoryNetworkContentTypes.EpochAccumulator,
-            value: { chainId: 1, blockHash: epochRootHash },
-          })
-        )
-        const encodedEpoch = await lookup.startLookup()
-        if (!(encodedEpoch instanceof Uint8Array)) return
-        epoch = EpochAccumulator.deserialize(encodedEpoch)
+        const block = await this.getBlockByHash(blockHash, includeTransactions)
+        return block
+      } catch (err) {
+        this.logger(`getBlockByNumber error: ${(err as any).message}`)
       }
-
-      blockHash = toHexString(epoch[blockIndex].blockHash)
+    } else {
+      throw new Error(`Blockhash index lookup not implemented`)
```

Comment:
I'm not sure I like the idea of throwing here.  It's not an error if we don't find the block or if the number is too high for the accumulator. I think we should just return undefined.